src/Mvc/Mvc.Core/src/Infrastructure/IHttpResponseWriterFactory.cs

```diff
@@ -2,21 +2,31 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.IO;
+using System.IO.Pipelines;
 using System.Text;
 
 namespace Microsoft.AspNetCore.Mvc.Infrastructure
 {
     /// <summary>
-    /// Creates <see cref="TextWriter"/> instances for writing to <see cref="Http.HttpResponse.Body"/>.
+    /// Creates <see cref="TextWriter"/> instances for writing to <see cref="Http.HttpResponse.Body"/> or <see cref="Http.HttpResponse.BodyWriter"/>.
     /// </summary>
-    public interface IHttpResponseStreamWriterFactory
+    public interface IHttpResponseWriterFactory
```

Comment:
Really the whole reason for this API was to abstract away the memory pool. Another option is to just *don't* when it comes to pipes. This change could be much more minimal if it didn't try to update this.

Reply:
Agree. I started without but later I brought it back to have similar abstractions with the same level of indirection. I don't think there was a technical reason to use this, but I could be wrong as it was a while back. I'll try to remove it.